docs/proposals/202005_scalable-rule-storage.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Scalable rule storage
+type: proposal
+menu: proposals
+status: proposed
+---
+
+## Summary
+
+There is no way to scale rule evaluation and storage today except functionally
+sharding rules onto multiple instances the `thanos rule` component. However, we
+have already solved scaling storage of time-series across multiple processes:
+`thanos receive`.
+
+To scale storing results of rule evaluations this proposal proposes to allow
+the `thanos rule` component to have a stateless mode, storing results of
+queries by sending it to a `thanos receive` hashring instead of storing them
+locally.
+
+## Motivation
+
+A few large rules can create a significant amount of resulting time-series,
+which should not be limited to what we can scale to with a single TSDB embedded
+in the rule component.
+
+## Goals
+
+Allow scaling storage of rule evaluations.
+
+## Verification
+
+* Run all rule component e2e tests with new mode as well.
+
+## Proposal
+
+Allow specifying one of the following flags:
+
+* `--remote-storage`
```

Comment:
Can we be more specifc? I think remote write? Also maybe we could use new gRPC based remote write for consistency? :thinking: 

The counter argument is that with normal remote write one can use any other system and deploy Thanos Ruler there (:

Reply:
I think we should start with the gRPC based one, and we can then as a second phase work on "generic" remote write

Reply:
Any suggestion on a better flag name?

Reply:
`remote-write` would do as we technically ask to push somewhere.

I think `remote-storage` is fine as well (: Not a biggie

Reply:
renamed to remote-write